proofs/tree.go

```diff
@@ -138,21 +137,29 @@ import (
 	"strconv"
 	"strings"
 
-	"regexp"
-
 	"github.com/centrifuge/precise-proofs/proofs/proto"
+	"github.com/ethereum/go-ethereum/common/hexutil"
 	"github.com/golang/protobuf/descriptor"
 	"github.com/golang/protobuf/proto"
 	"github.com/golang/protobuf/ptypes"
 	"github.com/golang/protobuf/ptypes/timestamp"
-	"github.com/iancoleman/strcase"
 	"github.com/xsleonard/go-merkle"
 )
 
 // DefaultSaltsLengthSuffix is the suffix used to store the length of slices (repeated) fields in the tree. It can be
 // customized with the SaltsLenghtSuffix TreeOption
 const DefaultSaltsLengthSuffix = "Length"
 
+type defaultValueEncoder struct{}
+
+func (valueEncoder *defaultValueEncoder) encodeToString(value []byte) string {
+	return hexutil.Encode(value)
+}
+
+type ValueEncoder interface {
+	encodeToString([]byte) string
```

Comment:
`encodeToString` must be exported else no one can implement the interface outside the package